shell/models/apps.deployment.js

```diff
@@ -19,7 +20,8 @@ export default class Deployment extends Workload {
         value: {
           metadata: {
             creationTimestamp: null,
-            labels:            { 'workload.user.cattle.io/workloadselector': revision.spec.template.metadata.labels['workload.user.cattle.io/workloadselector'] }
+            labels:            pickBy(revision.spec.template.metadata.labels, (value, key) => key !== 'pod-template-hash'),
+            annotations:       { 'cattle.io/timestamp': revision.spec.template.metadata.annotations['cattle.io/timestamp'] },
```

Comment:
out of interest do we need this, what happens if the timestamp annotation  isn't there in the revision

Reply:
If there is no timestamp, the normal old Pod will be removed during rollback.
To Reproduce
1. Create Deployment app, replica 1, image busybox, stdin yes, tty.
2. Click Redeploy on the UI, cattle.io/timestamp will be generated in spec.template.metadata.annotations.
3. Edit Deployment app，image busybox，no stdin yes，no tty。
4. At this time, the new Pod of the Deployed app cannot be created successfully, and the old Pod is running.
5. Rollback Deployment app，the old running Pod is also deleted.

There are still some problems in this way. I have made some adjustments.For spec.template.metadata.annotations, need to copy other items, refer to https://github.com/kubernetes/kubectl/blob/master/pkg/polymorphichelpers/rollback.go#L99-L166
The annotationsToSkip needs to be ignored, and the other annos need to be copied